src/components/new-safe/steps/Step0/index.tsx

```diff
@@ -0,0 +1,104 @@
+import { Box, Button, Divider, Grid, Paper, Typography } from '@mui/material'
+import useWallet from '@/hooks/wallets/useWallet'
+import ChainSwitcher from '@/components/common/ChainSwitcher'
+import useOnboard, { type ConnectedWallet, connectWallet } from '@/hooks/wallets/useOnboard'
+import NetworkSelector from '@/components/common/NetworkSelector'
+import useIsWrongChain from '@/hooks/useIsWrongChain'
+import { useCurrentChain } from '@/hooks/useChains'
+import { isPairingSupported } from '@/services/pairing/utils'
+
+import type { NewSafeFormData } from '@/components/new-safe/CreateSafe'
+import type { StepRenderProps } from '@/components/new-safe/CardStepper/useCardStepper'
+import KeyholeIcon from '@/components/common/icons/KeyholeIcon'
+import { OVERVIEW_EVENTS, trackEvent } from '@/services/analytics'
+import PairingQRCode from '@/components/common/PairingDetails/PairingQRCode'
+import PairingDescription from '@/components/common/PairingDetails/PairingDescription'
+
+export const ConnectWalletContent = ({
+  wallet,
+  isWrongChain,
+  setStep,
+}: {
+  wallet: ConnectedWallet | null
+  isWrongChain: boolean
+  setStep: StepRenderProps<NewSafeFormData>['setStep']
+}) => {
+  const chain = useCurrentChain()
```

Comment:
Why do we pass the connected wallet here?
```suggestion
  isWrongChain,
  setStep,
}: {
  isWrongChain: boolean
  setStep: StepRenderProps<NewSafeFormData>['setStep']
}) => {
  const wallet = useWallet()
  const chain = useCurrentChain()
```

Reply:
Because `wallet` is also used in the parent component but after looking at it again we can actually reuse the `useSafeCreation` there and access `wallet` and `isWrongChain` in this component directly like you suggest. Thanks!